worlds/hades/Client.py

```diff
@@ -0,0 +1,409 @@
+        hades_settings_string += str(self.hades_slot_data["underworld_customs_pact_amount"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["forced_overtime_pact_amount"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["heightened_security_pact_amount"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["routine_inspection_pact_amount"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["damage_control_pact_amount"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["approval_process_pact_amount"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["tight_deadline_pact_amount"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["personal_liability_pact_amount"]) + "-"
+                
+        #Codify in the string all the fillers values
+        hades_settings_string += str(self.hades_slot_data["darkness_pack_value"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["keys_pack_value"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["gemstones_pack_value"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["diamonds_pack_value"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["titan_blood_pack_value"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["nectar_pack_value"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["ambrosia_pack_value"]) + "-"
+
+        #Codify in the string all game settings
+        hades_settings_string += str(self.hades_slot_data["location_system"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["reverse_order_em"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["keepsakesanity"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["weaponsanity"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["storesanity"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["initial_weapon"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["ignore_greece_deaths"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["fatesanity"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["hidden_aspectsanity"]) + "-"
+        hades_settings_string += str(self.polycosmos_version) + "-"
+        hades_settings_string += str(self.hades_slot_data["automatic_rooms_finish_on_hades_defeat"]) + "-"
+
+        #Codify in the string all the finishing conditions
+        hades_settings_string += str(self.hades_slot_data["hades_defeats_needed"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["weapons_clears_needed"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["keepsakes_needed"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["fates_needed"]) + "-"
+            
+        #Store the compact setting string
+        self.compact_setting_string = hades_settings_string
```

Comment:
This is another example of something that I don't really think needs to be cached. Might as well just return `hades_settings_string` and replace your one use of `self.compact_setting_string` with a call to this function.